### PR TITLE
Adding IS Analytics LineChart

### PR DESCRIPTION
**Pull request by Ambujaa10**

### CODE REVIEW

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/src/LineChart.jsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import React, {Component} from 'react';
```
Comment (lasanthaS) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/src/LineChart.jsx]: Please add license header

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/pom.xml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+ <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+     <artifactId>wso2sp-solutions.is-analytics.is-analytics-lineChart</artifactId>
+    <packaging>pom</packaging>
+    <version>2.0.399-SNAPSHOT</version>
```
Comment (lasanthaS) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/pom.xml]: This should be 1.0.0-SNAPSHOT

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/pom.xml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+     <name>WSO2 Carbon Dashboards - IS Analytics - Widgets - LineChart</name>
+    <url>http://wso2.org</url>
+     <parent>
+        <groupId>org.wso2.carbon.analytics</groupId>
```
Comment (lasanthaS) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/pom.xml]: groupId should be org.wso2.analytics.solutions

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/SessionCount/pom.xml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+ <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+     <artifactId>wso2sp-solutions.is-analytics.is-analytics-sessionCount</artifactId>
+    <packaging>pom</packaging>
+    <version>2.0.399-SNAPSHOT</version>
```
Comment (dilini-muthumala) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/SessionCount/pom.xml]: remove this element.

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/TopLongestSession/pom.xml [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+ <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+     <artifactId>wso2sp-solutions.is-analytics.is-analytics-longestSession</artifactId>
+    <packaging>pom</packaging>
+    <version>2.0.399-SNAPSHOT</version>
```
Comment (dilini-muthumala) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/TopLongestSession/pom.xml]: remove this.

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/src/AverageSessionDuration.jsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+            legend: false,
+            append: false,
+        };
+        this.metadata = {
+               names: ['username', 'duration'],
+               types: ['ordinal', 'linear']
+
+        };
+
+        this.state ={
```
Comment (dilini-muthumala) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/src/AverageSessionDuration.jsx]: styling issues

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/src/LineChart.jsx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+import VizG from 'react-vizgrammar';
+import Widget from '@wso2-dashboards/widget';
+import {MuiThemeProvider, darkBaseTheme, getMuiTheme} from 'material-ui/styles';
+
+class LineChart extends Widget{
```
Comment (TanyaM) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/src/LineChart.jsx]: Can't we rename the LineChart with its actual usage?
ex: AverageSessionDuration

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/Messages/pom.xml [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+ <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+     <artifactId>wso2sp-solutions.is-analytics.is-analytics-messages</artifactId>
+    <packaging>pom</packaging>
+    <version>2.0.399-SNAPSHOT</version>
```
Comment (TanyaM) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/Messages/pom.xml]: Incorrect version

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/Messages/src/Messages.jsx [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        {
+        
+        for(let j=0;j<message.data.length;j++)
+        { for(let i=0;i<10;i++){
+            if (i==5){
+            switch(message.data[j][i]){
+                case 0: 
+                message.data[j][i]='False';break;
+                case 1: 
+                message.data[j][i]='True';break;
```
Comment (TanyaM) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/Messages/src/Messages.jsx]: Above snippet is not properly formatted.

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/SessionCount/pom.xml (hunk 4 above)
Comment (TanyaM) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/SessionCount/pom.xml]: Incorrect version

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/TopLongestSession/pom.xml (hunk 5 above)
Comment (TanyaM) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/TopLongestSession/pom.xml]: Incorrect version

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/package.json [hunk 10]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "AverageSessionDuration",
```
Comment (Niveathika92) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/package.json]: Name should be as per the standard "Is Analytics Average Session Duration", while id is "IsAnalyticsAverageSessionDuration". Do the change in directory name as well

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/src/AverageSessionDuration.jsx [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
```
Comment (Niveathika92) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/src/AverageSessionDuration.jsx]: import react as best practice.

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/package.json [hunk 12]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "LineChart",
```
Comment (Niveathika92) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/LineChart/package.json]: Lets rename it to something meaningful "IsAnalyticsSessionCountOverTime"

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/src/resources/widgetConf.json [hunk 13]
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "AverageSessionDuration",
```
Comment (Niveathika92) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/AverageSessionDuration/src/resources/widgetConf.json]: Name should be as per the standard "Is Analytics Average Session Duration", while id is "IsAnalyticsAverageSessionDuration". Do the change in directory name as well

**Pull request merged**